SWUTILS-903: Highlight X3 hybrid and PV variants

dchadwic-xilinx opened the pull request:
As a test I used the device id of the X3 NIC. Before I merge this I wanna test on the real pair of variants. Raising the PR now because I want my methodology critiqued. I also will use this commit to add an extension to SWUTILS-833.
When highlighted the cell lights up like this.
![image](https://github.com/user-attachments/assets/9efe6eec-0f1e-4242-b1a8-55f40412b70c)
![image](https://github.com/user-attachments/assets/5ad4316f-4b5a-44ff-a364-a4611b487396)

ibeecraft-amd commented:
As mentioned in a quick chat. 
It would be nice to extend the interesting value functionality to allow for custom messages. Highlighting whether it is a Hybrid or PV at a glance rather than just outputting the check that failed. I know for certain I will forget how to decode the device ID.

dchadwic-xilinx commented:
Turns out we only save a PCI device as a Xilinx PCI device if the PCI device class = 200 which corresponds to network controllers. Now this change saves all devices which match the PV and hybrid device IDs to a new table:
![image](https://github.com/user-attachments/assets/e12b4dbe-cca4-47fa-847a-3c168aa6e026)
![image](https://github.com/user-attachments/assets/5caf544c-f01c-40a6-992b-33486eed4278)
This table could be extended to highlight other devices we may be interested in. Note that the ability to use the sfreport lspci script still exists if needed.

While doing this. I also fixed-up the ability to compare a key's value to another key's value. As a demonstation of this I added a commit which highlights when ctpio_erase != ctpio_poison (poisoned frames have been transmitted).
![image](https://github.com/user-attachments/assets/422364ae-2c41-44c3-9db2-aff42ff660f1)
![image](https://github.com/user-attachments/assets/7c91951e-76d9-4058-b64a-c76f500badf4)

ibeecraft-amd reviewed (changes requested):
Happy to approve once minor changes to indentation fixed. Some places in sfreport have poor indentation but we should aim to be better going forward.
